src/clr-angular/forms/combobox/model/multi-select-combobox.model.ts

```diff
@@ -0,0 +1,88 @@
+/**
+ * Copyright (c) 2016-2020 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { ComboboxModel } from './combobox.model';
+
+export class MultiSelectComboboxModel<T> implements ComboboxModel<T> {
+  model: T[];
+
+  containsItem(item: T): boolean {
+    return this.model ? this.model.indexOf(item) > -1 : false;
+  }
+
+  setSelected(item: T, selected: boolean): void {
+    if (selected) {
+      this.addItem(item);
+    } else {
+      this.removeItem(item);
+    }
+  }
+
+  pop(): T {
+    let item;
+    if (this.model && this.model.length > 0) {
+      item = this.model[this.model.length - 1];
+      this.removeItem(item);
+    }
+    return item;
+  }
+
+  toString(displayField?: string, index: number = -1): string {
+    let displayString: string = '';
+
+    if (this.model) {
+      if (index > -1) {
+        if (this.model[index]) {
+          if (displayField && this.model[index][displayField]) {
+            displayString += this.model[index][displayField];
+          } else {
+            displayString += this.model[index].toString();
+          }
+        }
+      } else {
+        this.model.forEach((model: T) => {
+          if (displayField && model[displayField]) {
+            displayString += model[displayField];
+          } else {
+            displayString += model.toString();
+          }
+          displayString += ' ';
+        });
+      }
+    }
+
+    return displayString.trim();
+  }
+
+  private addItem(item: T) {
+    if (!this.model) {
+      this.model = [];
+    }
+
+    const index = this.model.indexOf(item);
+
+    if (index === -1) {
+      this.model.push(item);
+    }
+  }
+
```

Comment:
```ts
private addItem(item: T) {
  // Check to see that is not already there if not
  if (!this.containsItem(item)) {
    // integrate the first if here, make sure that this.model will be array and add the new item into it.
    this.model = (this.model || []).push(item);
  }
}

Reply:
I like it. Though you can't combine the assignment with the .push() in one line.
I'll still use it. Calling the this.containsItem() helps it be more concise and clear to understand. Thanks